### PR TITLE
Fixes #19038: Add a relay technique

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19038

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: techniques/system/rudderRelay/1.0/apache/system_rudder_apache_webdav_configuration.cf [hunk 1]
```diff
@@ -0,0 +1,37 @@
+bundle agent system_rudder_apache_webdav_configuration {
+  vars:
+    SuSE::
```
Comment (amousset) [inline techniques/system/rudderRelay/1.0/apache/system_rudder_apache_webdav_configuration.cf:3]: we should probably use `suse` everywhere for consistency

File: techniques/system/rudderRelay/1.0/apache/main.cf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+bundle common rudder_apache {
```
Comment (amousset) [inline techniques/system/rudderRelay/1.0/apache/main.cf:1]: I don't think relayd and apache should be the same technique

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

File: techniques/system/rudderRelay/1.0/relayd/relayd.cf [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  vars:
+      "config_dir"          string => "${g.rudder_base}/etc/relayd";
+      "config_file"         string => "${config_dir}/main.conf";
+      "config_log_file"     string => "${config_dir}/logging.conf";
+      "service_name"        string => "rudder-relayd";
+}
+bundle agent system_rudder_relay_configuration {
+  vars:
+      "config_class_prefix" string => canonify("file_from_template_${rudder_relayd.config_file}");
+      "component"           string => "Rudder-relayd service";
```
Comment (amousset) [inline techniques/system/rudderRelay/1.0/relayd/relayd.cf:28]:
```suggestion
      "component"           string => "rudder-relayd service";
```

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

**Review by amousset — APPROVED** [review]

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**